File: Sources/FitUI/Components/FitEarningsHero.swift
import SwiftUI

// MARK: - FitEarningsHero
//
// Balance / earnings hero card — large amount + period label,
// optional trend pill, optional 60/40 breakdown grid, optional
// "planned" footer line, and an empty-state shape. Used at the
// top of balance.html and the coach earnings dashboard.
// See `docs/components.md` § FitEarningsHero.

public enum FitEarningsTrend {
    case up(percent: Double)
    case down(percent: Double)
    case flat
}

public struct FitEarningsHeroBreakdown {
    public let label: String
    public let value: String

    public init(label: String, value: String) {
        self.label = label
        self.value = value
    }
}

public struct FitEarningsHero: View {
    let amount: String
    let period: String
    let trend: FitEarningsTrend?
    let breakdown: [FitEarningsHeroBreakdown]
    let planned: String?
    let isEmpty: Bool

    @Environment(\.fitTheme) private var theme

    public init(
        amount: String,
        period: String,
        trend: FitEarningsTrend? = nil,
        breakdown: [FitEarningsHeroBreakdown] = [],
        planned: String? = nil,
        isEmpty: Bool = false
    ) {
        self.amount = amount
        self.period = period
        self.trend = trend
        self.breakdown = breakdown
        self.planned = planned
        self.isEmpty = isEmpty
    }

    public var body: some View {
        VStack(alignment: .leading, spacing: FitSpacing.sp3) {
            // Header
            HStack(alignment: .firstTextBaseline) {
                VStack(alignment: .leading, spacing: 2) {
                    Text(isEmpty ? "—" : amount)
                        .font(.custom(FitFont.family, size: 28).weight(.semibold))
                        .foregroundColor(theme.textPrimary)
                    Text(period)
                        .font(.custom(FitFont.family, size: 13))
                        .foregroundColor(theme.textTertiary)
                }
                Spacer()
                if let trend, !isEmpty {
                    trendPill(trend)
                }
            }

            if !breakdown.isEmpty && !isEmpty {
                Rectangle()
                    .fill(theme.divider)
                    .frame(height: 1)

                HStack(alignment: .top, spacing: FitSpacing.sp3) {
                    if breakdown.count >= 1 {
                        breakdownColumn(breakdown[0])
                            .frame(maxWidth: .infinity, alignment: .leading)
                            .layoutPriority(60)
                    }
                    if breakdown.count >= 2 {
                        breakdownColumn(breakdown[1])
                            .frame(maxWidth: .infinity, alignment: .leading)
                            .layoutPriority(40)
                    }
                }
            }

            if let planned, !isEmpty {
                Text(planned)
                    .font(.custom(FitFont.family, size: 13))
                    .foregroundColor(theme.textTertiary)
            }
        }
        .padding(FitSpacing.sp4)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(theme.surfaceHigh)
        .clipShape(RoundedRectangle(cornerRadius: FitRadius.lg))
    }

    private func breakdownColumn(_ row: FitEarningsHeroBreakdown) -> some View {
        VStack(alignment: .leading, spacing: 2) {
            Text(row.label)
                .font(.custom(FitFont.family, size: 13))
                .foregroundColor(theme.textTertiary)
            Text(row.value)
                .font(.custom(FitFont.family, size: 16).weight(.medium))
                .foregroundColor(theme.textPrimary)
        }
    }

    @ViewBuilder
    private func trendPill(_ trend: FitEarningsTrend) -> some View {
        switch trend {
        case .up(let pct):
            FitBadge(format(pct, prefix: "↑ "), style: .success)
        case .down(let pct):
            FitBadge(format(pct, prefix: "↓ "), style: .danger)
        case .flat:
            FitBadge("—", style: .neutral)
        }
    }

    private func format(_ pct: Double, prefix: String) -> String {
        let abs = abs(pct)
        let str = String(format: "%.0f", abs)
        return "\(prefix)\(str)%"
    }
}

File: android/src/main/kotlin/com/fit321/fitui/components/FitEarningsHero.kt
package com.fit321.fitui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.fit321.fitui.theme.LocalFitTheme
import com.fit321.fitui.tokens.FitFont
import com.fit321.fitui.tokens.FitRadius
import com.fit321.fitui.tokens.FitSpacing
import kotlin.math.abs

// ============================================================================
// FitEarningsHero — hero summary card on balance / earnings.
// Mirrors Swift FitEarningsHero. See `docs/components.md` § FitEarningsHero.
// ============================================================================

sealed class FitEarningsTrend {
    data class Up(val percent: Double) : FitEarningsTrend()
    data class Down(val percent: Double) : FitEarningsTrend()
    object Flat : FitEarningsTrend()
}

data class FitEarningsHeroBreakdown(
    val label: String,
    val value: String
)

@Composable
fun FitEarningsHero(
    amount: String,
    period: String,
    trend: FitEarningsTrend? = null,
    breakdown: List<FitEarningsHeroBreakdown> = emptyList(),
    planned: String? = null,
    isEmpty: Boolean = false,
    modifier: Modifier = Modifier
) {
    val theme = LocalFitTheme.current

    Column(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(FitRadius.lg))
            .background(theme.surfaceHigh, RoundedCornerShape(FitRadius.lg))
            .padding(FitSpacing.sp4),
        verticalArrangement = Arrangement.spacedBy(FitSpacing.sp3)
    ) {
        // Header
        Row(verticalAlignment = Alignment.Top) {
            Column(modifier = Modifier.weight(1f)) {
                Text(
                    if (isEmpty) "—" else amount,
                    style = FitFont.headline.copy(fontSize = 28.sp, fontWeight = FontWeight.SemiBold),
                    color = theme.textPrimary
                )
                Text(
                    period,
                    style = FitFont.footnote,
                    color = theme.textTertiary
                )
            }
            if (trend != null && !isEmpty) {
                trendBadge(trend)
            }
        }

        if (breakdown.isNotEmpty() && !isEmpty) {
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(1.dp)
                    .background(theme.divider)
            )
            Row(
                horizontalArrangement = Arrangement.spacedBy(FitSpacing.sp3),
                verticalAlignment = Alignment.Top
            ) {
                if (breakdown.size >= 1) {
                    BreakdownColumn(breakdown[0], modifier = Modifier.weight(0.6f))
                }
                if (breakdown.size >= 2) {
                    BreakdownColumn(breakdown[1], modifier = Modifier.weight(0.4f))
                }
            }
        }

        if (planned != null && !isEmpty) {
            Text(
                planned,
                style = FitFont.footnote,
                color = theme.textTertiary
            )
        }
    }
}

@Composable
private fun BreakdownColumn(row: FitEarningsHeroBreakdown, modifier: Modifier = Modifier) {
    val theme = LocalFitTheme.current
    Column(modifier = modifier) {
        Text(row.label, style = FitFont.footnote, color = theme.textTertiary)
        Text(
            row.value,
            style = FitFont.body1.copy(fontWeight = FontWeight.Medium),
            color = theme.textPrimary
        )
    }
}

@Composable
private fun trendBadge(trend: FitEarningsTrend) {
    when (trend) {
        is FitEarningsTrend.Up   -> FitBadge("↑ ${formatPct(trend.percent)}", FitBadgeStyle.Success)
        is FitEarningsTrend.Down -> FitBadge("↓ ${formatPct(trend.percent)}", FitBadgeStyle.Danger)
        FitEarningsTrend.Flat    -> FitBadge("—", FitBadgeStyle.Neutral)
    }
}

private fun formatPct(p: Double): String = "${abs(p).toInt()}%"
